test/config_cat/cache_policy/auto_test.exs

```diff
@@ -153,56 +98,74 @@ defmodule ConfigCat.CachePolicy.AutoTest do
       assert {:ok, settings, entry.fetch_time_ms} == CachePolicy.get(instance_id)
     end
 
-    test "calls the change callback", %{entry: entry} do
-      test_pid = self()
-      %{entry: old_entry} = make_old_entry()
-      expect_refresh(old_entry)
+    test "does not update config when server responds that the config hasn't changed", %{
+      entry: entry
+    } do
+      expect_refresh(entry)
+      {:ok, instance_id} = start_cache_policy(@policy)
 
-      policy = CachePolicy.auto(on_changed: fn -> send(test_pid, :callback) end)
-      {:ok, instance_id} = start_cache_policy(policy)
+      expect_unchanged()
 
-      assert_receive(:callback)
+      assert :ok = CachePolicy.force_refresh(instance_id)
+    end
 
+    @tag capture_log: true
+    test "handles error responses", %{entry: entry} do
       expect_refresh(entry)
-      :ok = CachePolicy.force_refresh(instance_id)
+      {:ok, instance_id} = start_cache_policy(@policy)
 
-      assert_receive(:callback)
+      assert_returns_error(fn -> CachePolicy.force_refresh(instance_id) end)
     end
+  end
 
-    @tag capture_log: true
-    test "handles errors in the change callback", %{entry: entry, settings: settings} do
```

Comment:
This test was removed; the exception-handling behavior of hooks is well-tested elsewhere.